### PR TITLE
Add SCSt app common micrometer tags

### LINKED ISSUE #35: Add Micrometer tags to discriminate streams, apps and instances
Create an auto-configuration to inject common tags for: 
- stream name
- application name
- application instance
- application guid
- application type

### PR DESCRIPTION
**Pull request by tzolov**
Resolve #35

### CODE REVIEW

File: common/app-starters-micrometer-common/src/main/java/org/springframework/cloud/stream/app/micrometer/common/SpringCloudStreamMicrometerCommonTags.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	private String applicationGuid;
+
+	@Value("${spring.cloud.dataflow.stream.app.type:unknown}")
+	private String applicationType;
+
+	@Bean
+	public MeterRegistryCustomizer<MeterRegistry> metricsCommonTags() {
+		return registry -> registry.config()
+				.commonTags("clusterName", clusterName)
+				.commonTags("stream", streamName)
```
Comment (olegz) [inline common/app-starters-micrometer-common/src/main/java/org/springframework/cloud/stream/app/micrometer/common/SpringCloudStreamMicrometerCommonTags.java]: To be consistent, should it be `streamName`?

**Comment by jvalkeal** [comment]
If this is supposed to go to all apps, dep should be added to as it's a bit too much to modify all apps, no?

https://github.com/spring-cloud-stream-app-starters/core/blob/31b8e65e56f484c3d822675844809205c83e6a95/pom.xml#L47

File: common/app-starters-micrometer-common/src/main/java/org/springframework/cloud/stream/app/micrometer/common/SpringCloudStreamMicrometerCommonTags.java (hunk 1 above)
Comment (tzolov) [reply common/app-starters-micrometer-common/src/main/java/org/springframework/cloud/stream/app/micrometer/common/SpringCloudStreamMicrometerCommonTags.java]: I guess it should. Will change it

**Comment by tzolov** [comment]
Adding the dependency to the core/pom lead to cyclic dependency error. This is because i've made the `app-starters-micrometer-common` inherit from `app-starters-common` which in turn inherits from `app-starters-build` (e.g. the core\pom). So the `app-starters-micrometer-common` is child of the core\pom and is added as dependency as well. 
Perhaps i should l make `app-starters-micrometer-common` inherit from the `spring-boot-starters` instead, as it was in the POC. @jvalkeal , @sobychacko any thoughts on this?

**Comment by sobychacko** [comment]
@tzolov I didn't think about that. The problem is that these common artifacts are not necessarily designed to be common across the entire spectrum of app starters to be used in core. They are picked individually by downstream starters as needed. If we are going to bring this to core pom, then we can't have this cyclic dependency. What I suggest is instead of adding this to core as a dependency add it as a configuration to the maven plugin: https://github.com/spring-cloud-stream-app-starters/core/blob/master/pom.xml#L177
This way, we will have the same effect - this artifact will be added to each of the generated app as a dependency.

**Comment by jvalkeal** [comment]
well lol, maybe this was a wrong location for that module. do we have any better place without adding yet another repo and waffle ;)

**Comment by jvalkeal** [comment]
maybe we have a leeway to use something what soby mentioned as this is a temporary fix and will eventually go away when generic feature is added to boot itself.

**Comment by sobychacko** [comment]
@tzolov if someone is generating from scratch, then I don't think the cyclic dependency issue will be there, since as you mentioned it is inheriting from `spring-boot-starter-parent`.

**Comment by sobychacko** [comment]
LGTM - merged.

**Pull request closed without merging**